swagger_spec_validator/common.py

```diff
@@ -23,9 +28,78 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def load_json(url):
-    with contextlib.closing(request.urlopen(url, timeout=TIMEOUT_SEC)) as fh:
-        return json.loads(fh.read().decode('utf-8'))
+def _is_yaml(uri, content_type=None):
+    yaml_content_types = {'application/yaml', 'application/x-yaml', 'text/yaml'}
+    yaml_file_extensions = {'.yaml', '.yml'}
+
+    if content_type in yaml_content_types:
+        return True
+
+    _, ext = os.path.splitext(uri)
+    if ext.lower() in yaml_file_extensions:
+        return True
+
+    return False
+
+
+def _is_json(uri, content_type=None):
+    json_content_types = {
+        'application/json', 'application/x-javascript',
+        'text/javascript', 'text/x-javascript', 'text/x-json',
+    }
+    json_file_extensions = {'.json'}
+
+    if content_type in json_content_types:
+        return True
+
+    _, ext = os.path.splitext(uri)
+    if ext.lower() in json_file_extensions:
+        return True
+
+    return False
+
+
+def _read_file_pointer(fp, is_json, is_yaml):
+    assert not (is_json and is_yaml), 'fp cannot be JSON and YAML at the same time'
+
+    if is_json:
+        return json.load(fp)
+
+    if not is_yaml:
+        warnings.warn(
+            message='fp is not recognized as JSON and YAML. Fallback to YAML reading',
+            category=Warning,
+        )
+
+    return yaml.safe_load(fp)
+
+
+def read_file(file_path):
+    """
+    Utility method for reading a JSON/YAML file and converting it to a Python dictionary
+    :param file_path: path of the file to read
+
+    :return: Python dictionary representation of the JSON file
+    :rtype: dict
+    """
+    with open(file_path) as fp:
+        return _read_file_pointer(fp, is_json=_is_json(file_path), is_yaml=_is_yaml(file_path))
+
+
+def read_url(url):
+    parsed_url = urlparse(url)
+    if parsed_url.scheme == 'file':
+        fp = urlopen(url)
+        is_json = _is_json(parsed_url.path)
+        is_yaml = _is_yaml(parsed_url.path)
+        return _read_file_pointer(fp, is_json=is_json, is_yaml=is_yaml)
+
+    with contextlib.closing(request.urlopen(url, timeout=TIMEOUT_SEC)) as remote_request:
+        # NOTE: JSON is a subset of YAML so it is safe to read JSON as it is YAML
```

Comment:
@sjaensch are you proposing to revert to read JSON and YAML as they're YAML files?
Asking because current tests failure are related to this ... so I can avoid to invest additional time in fixing them :P